gateway/modules/auth/match.go

```diff
@@ -94,11 +94,11 @@ func (m *Module) matchQuery(ctx context.Context, project string, rule *config.Ru
 	// Execute the read request
 	data, err := crud.Read(ctx, rule.DB, project, rule.Col, req)
 	if err != nil {
-		return nil, err
+		return nil, formaterror(rule, err)
 	}
 	err = utils.StoreValue("args.result", data, args)
 	if err != nil {
-		return nil, err
+		return nil, formaterror(rule, err)
 	}
 	return m.matchRule(ctx, project, rule.Clause, args, nil)
```

Comment:
Pass the error returned by match rule inside format error. Return the postAction object as is